### PR TITLE
[TACHYON-1241] Re-enable ignored tests

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses https://tachyon.atlassian.net/browse/TACHYON-1241

### CODE REVIEW

File: integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java [hunk 1]
```diff
@@ -69,4 +70,17 @@ public void tooManyColumnsTest() throws Exception {
     mRawTableMaster.createRawTable(new TachyonURI("/testTable"),
         mMasterConf.getInt(Constants.MAX_COLUMNS) + 1, null);
   }
+
+  @Test
+  public void nullMetadataTest() throws Exception {
+    mException.expect(NullPointerException.class);
```
Comment (yupeng9) [inline integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java]: Is `null` value expected as input? If not, perhaps add a precondition check with some exception message?

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java [hunk 2]
```diff
@@ -127,6 +127,12 @@ public void addTempBlockMeta(TempBlockMeta tempBlockMeta)
    */
   public void commitTempBlockMeta(TempBlockMeta tempBlockMeta)
       throws WorkerOutOfSpaceException, BlockAlreadyExistsException, BlockDoesNotExistException {
+    long blockId = tempBlockMeta.getBlockId();
+    if (hasBlockMeta(blockId)) {
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: Is this necessary? `dir.addBlockMeta` on line 139 already throws `BlockAlreadyExistsException`.

Comment (gpang) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: Was this an existing bug? What was the old behavior, and what is the new behavior with this code?

**Comment by gpang** [comment]
I just had a question about the new code in BlockMetadataManager.java. Thanks!

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java (hunk 2 above)
Comment (uronce-cc) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: It's a bug in my understanding. If `tempBlockMeta` has the same id as a committed block which is in another `StorageDir`, then `dir.addBlockMeta` won't throw `BlockAlreadyExistsException`. 

The new codes enable [this test](https://github.com/amplab/tachyon/pull/2106/files#diff-7332fdb1d95b81852823dda9dec454cfL429).

File: servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java [hunk 3]
```diff
@@ -424,18 +423,16 @@ public void moveTempBlockTest() throws Exception {
     mBlockStore.moveBlock(SESSION_ID1, TEMP_BLOCK_ID, mTestDir2.toBlockStoreLocation());
   }
 
-  // TACHYON-825
-  @Ignore
   @Test
-  public void moveBlockToTheLocationWithExistingIdTest() throws Exception {
+  public void cacheSameBlockInDifferentTiersTest() throws Exception {
```
Comment (uronce-cc) [inline servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java]: Is there a test case for caching the same block in different dirs of the same tier?

File: servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java [hunk 4]
```diff
@@ -424,18 +423,16 @@ public void moveTempBlockTest() throws Exception {
...
     mThrown.expect(BlockAlreadyExistsException.class);
-    mThrown.expectMessage("Failed to add BlockMeta: blockId " + BLOCK_ID1 + " exists");
+    mThrown.expectMessage(ExceptionMessage.ADD_EXISTING_BLOCK.getMessage(BLOCK_ID1,
+        FIRST_TIER_ALIAS));
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java]: better to put line 429 and 430 before the second `TieredBlockStoreTestUtils.cache`, so we know explicitly the second `cache` is the expected call to trigger this exception.

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java (hunk 2 above)
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: this looks like a bug, however, this may be affecting speculative execution. @calvinjia , is the original code intended?

File: integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java [hunk 5]
```diff
@@ -49,10 +51,9 @@ public final void before() throws Exception {
         mLocalTachyonClusterResource.get().getMaster().getInternalMaster().getFileSystemMaster();
   }
 
-  @Ignore
   @Test
   public void createRawTableTest() throws Exception {
-    mRawTableMaster.createRawTable(new TachyonURI("/testTable"), 1, null);
+    mRawTableMaster.createRawTable(new TachyonURI("/testTable"), 1, ByteBuffer.allocate(0));
```
Comment (apc999) [inline integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java]: do we also need to make similar change (`null`  -> `ByteBuffer.allocate(0)`) in this file?

Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java]: no, because the other calls are testing edge cases

File: integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java (hunk 1 above)
Comment (jsimsa) [inline integration-tests/src/test/java/tachyon/master/rawtable/RawTableMasterIntegrationTest.java]: Done.

File: servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java (hunk 2 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java]: I discussed this with @calvinjia offline and we concluded that the original code makes it possible to create a block with the same id in two different tiers. That is bad because a some functions in block manager assume that the id is unique (e.g. `getBlockMeta(id)` returns BlockMeta for the given id ... it iterate through storage dirs looking for the id until it finds it)

it should not be possible to create a block with the same id in different tiers.

File: servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java (hunk 3 above)
Comment (jsimsa) [inline servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java]: Turns out this was the test case (both mTestDir1 and mTestDir2 are in the same tier). I added another test that uses mTestDir1 and mTestDir3 that are in different tiers.

File: servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java (hunk 4 above)
Comment (jsimsa) [inline servers/src/test/java/tachyon/worker/block/TieredBlockStoreTests.java]: Done.

**Comment by jsimsa** [comment]
Thanks for the review guys, addressed your comments. PTAL

**Comment by jsimsa** [comment]
Jenkins, test this please.

**Comment by jsimsa** [comment]
ping

**Comment by gpang** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM

**Comment by apc999** [comment]
LGTM
